cargo-culture-kit/src/rules/mod.rs

```diff
@@ -56,18 +56,28 @@ pub trait Rule: Debug {
 
     /// Does the Rust project found at `cargo_manifest_path` uphold this
     /// `Rule`, as summarized in the `description`?
-    ///
-    ///
-    /// Pre-parsed cargo `metadata` may be available, and supplemental
-    /// human-readable `verbose` content may be written to the
-    /// `print_output`.
-    fn evaluate(
-        &self,
-        cargo_manifest_file_path: &Path,
-        verbose: bool,
-        metadata: &Option<Metadata>,
-        print_output: &mut Write,
-    ) -> RuleOutcome;
+    fn evaluate(&self, context: RuleContext) -> RuleOutcome;
+}
+
+/// Parameter struct for the `Rule::evaluate` method.
+/// Should provide the minimum information necessary for
+/// project-level quality and completeness checks to be run.
+pub struct RuleContext<'a> {
+    /// The path of the Cargo.toml project file for the Rust
+    /// project currently under evaluation.
+    pub cargo_manifest_file_path: &'a Path,
+    /// When true, `Rule` implementations should supply additional
+    /// human-reader-oriented content by writing to `print_output`
+    pub verbose: bool,
+    /// Pre-parsed cargo metadata for the current project under evaluation.
+    /// Ought to be `None` only when the cargo metadata retrieval or parsing
+    /// fails.
```

Comment:
The last part is strange. Are there some rules that can (and should) run even in absence of the metadata? Even if so, perhaps this could be simplified by requiring this step to succeed as a top-level check.

Reply:
Absolutely, some rules pass in the absence of project metadata. The easiest examples are the ones that check for the presence of useful files, e.g. `HasReadmeFile`.  I'd like to maximize the amount of useful information that can be extracted even in the face of a metadata problem, so that even if [CargoMetadataReadable](https://docs.rs/cargo-culture-kit/0.1.0/cargo_culture_kit/struct.CargoMetadataReadable.html) fails, the rest of the checks can still get a swing at the project.

Reply:
:metal: